plugins/out_es/es_bulk.c

```diff
@@ -77,13 +78,14 @@ int es_bulk_append(struct es_bulk *bulk, char *index, int i_len,
          *    1. rest of msgpack data size
          *    2. ratio from bulk json size and processed msgpack size.
          */
+        append_size = required - available;
         if (converted_size == 0) {
             /* converted_size = 0 causes div/0 */
             flb_debug("[out_es] converted_size is 0");
-            append_size = required - available;
         } else {
-            append_size = (whole_size - converted_size) /* rest of size to convert */
-                        * (bulk->size / converted_size); /* = json size / msgpack size */
+            int remaining_size = ceil((whole_size - converted_size) /* rest of size to convert */
+                * ((double)bulk->size / converted_size));           /* = json size / msgpack size */
+            append_size = max(append_size, remaining_size);
```

Comment:
I ma not sure these functions exist in math.h: https://man7.org/linux/man-pages/man0/math.h.0p.html

Looks like it may be called fmax?

```
../../library/libflb-plugin-out_es.a(es_bulk.c.o): In function `es_bulk_append':
/home/runner/work/fluent-bit/fluent-bit/plugins/out_es/es_bulk.c:88: undefined reference to `max'
collect2: error: ld returned 1 exit status
tests/runtime/CMakeFiles/flb-rt-in_cpu.dir/build.make:323: recipe for target 'bin/flb-rt-in_cpu' failed
make[2]: *** [bin/flb-rt-in_cpu] Error 1
CMakeFiles/Makefile2:6777: recipe for target 'tests/runtime/CMakeFiles/flb-rt-in_cpu.dir/all' failed
make[1]: *** [tests/runtime/CMakeFiles/flb-rt-in_cpu.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
../library/libflb-plugin-out_es.a(es_bulk.c.o): In function `es_bulk_append':
/home/runner/work/fluent-bit/fluent-bit/plugins/out_es/es_bulk.c:88: undefined reference to `max'
collect2: error: ld returned 1 exit status
```

Reply:
@gitfool

Reply:
I'll switch to `fmax` but I thought an integer `max` function was available as I saw it used elsewhere in the code.

Reply:
Looks like that's the windows code though, be careful, Fluent Bit is compiled on lots of platforms...

Reply:
Of course! That helps explain my confusion. 😊